docker/Dockerfile

```diff
@@ -138,6 +137,25 @@ ENV PATH="${PATH}:/usr/local/lib/nodejs/node-v17.9.0-linux-x64/bin"
 
 RUN npm install --global markdownlint-cli@0.31.1
 
+FROM dependencies as dependencies-dev-platform-arm64
+# Install hadolint for linting Dockerfiles
+RUN curl -Lo /usr/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.10.0/hadolint-Linux-arm64 && \
+    chmod +x /usr/bin/hadolint
+
+# Install NodeJS and use it to install markdownlint-cli
+
+RUN curl -Lo /tmp/node.tar.xz https://nodejs.org/dist/v17.9.0/node-v17.9.0-linux-arm64.tar.xz && \
+    mkdir -p /usr/local/lib/nodejs && \
+    tar -xf /tmp/node.tar.xz -C /usr/local/lib/nodejs && \
+    rm -rf /tmp/node.tar.xz
+
+ENV PATH="${PATH}:/usr/local/lib/nodejs/node-v17.9.0-linux-arm64/bin"
```

Comment:
These ENV statements don't work in a collapsed setup, unless I create symlinks.

`TARGETARCH` returns either `amd64` or `arm64`

NodeJS publishes their `amd64` compatible builds as `-x64`. For ARM they are the same, `-arm64`.

Because of this, and both NodeJS and Hadolint having different file paths, I believe the split intermediate image is better.